api/grpcserver/mesh_service.go

```diff
@@ -610,3 +614,68 @@ func (s MeshService) EpochStream(req *pb.EpochStreamRequest, stream pb.MeshServi
 	)
 	return nil
 }
+
+func (s MeshService) MalfeasanceQuery(ctx context.Context, req *pb.MalfeasanceRequest) (*pb.MalfeasanceResponse, error) {
+	var id types.NodeID
+	if len(req.SmesherHex) == 0 {
```

Comment:
See my comment in the API PR: I think we shouldn't have to fields in the request encoding the same information.